src/bootstrap/builder.rs

```diff
@@ -618,6 +618,10 @@ impl<'a> Builder<'a> {
         // FIXME: cargo bench does not accept `--release`
         if self.config.rust_optimize && cmd != "bench" {
             cargo.arg("--release");
+
+            if mode != Mode::Libstd {
+                cargo.env("RUSTC_THINLTO", "1");
```

Comment:
Why do we not want this for std?

This will (afaict) override the codegen-units setting in config.toml unconditionally, and isn't gated either, both of which seem somewhat odd. I think I'm okay with the latter (we expect ThinLTO to work reliably and not lose performance) but I'd prefer that we then remove codegen-units from the configuration.

OTOH, if we are disabling this for std due to performance concerns, then I feel like it should be behind a toggle.

Reply:
At the moment I'm just trying to be conservative, my plan is to, after this lands, enable this for libstd. I just suspect that'll have more bugs/regressions associated with it (for whatever reason) and wanted to get the rustc wins ASAP